src/main/java/com/example/controller/SimulationController.java

```diff
@@ -252,12 +260,33 @@ private void processStep() {
         statisticsController.addStats(report.getNumSupporting(), report.getNumNotSupporting());
     }
 
+    public void openResultDialog() {
+        Platform.runLater(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    FXMLLoader fxmlLoader = new FXMLLoader();
+                    fxmlLoader.setLocation(getClass().getResource("/view/simulationResultView.fxml"));
+                    Scene scene = new Scene(fxmlLoader.load());
+                    ((SimulationResultController)fxmlLoader.getController()).setMessage(graphController.getGraph().checkConsensus());
+                    Stage stage = new Stage();
+                    stage.setTitle("Result");
+                    stage.setScene(scene);
+                    stage.show();
+                } catch (IOException ioException) {
+                    ioException.printStackTrace();
+                }
+            }
+        });
+    }
+
     public void doStepTask() {
         if (!isFinished.get()) {
             processStep();
 
             if (isFinished.get()) {
                 System.out.println("Finished");
+                openResultDialog();
```

Comment:
We call this function in 3 tasks, maybe creating a common function e.g. onFinished() would be a good idea?